nilearn/decomposition/_subject_pca.py

```diff
@@ -0,0 +1,330 @@
+        Used to cache the masking process.
+        By default, no caching is done. If a string is given, it is the
+        path to the caching directory.
+
+    memory_level: integer, optional
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    n_jobs: integer, optional
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs', -2 'all CPUs but one', and so on.
+
+    verbose: integer, optional
+        Indicate the level of verbosity. By default, nothing is printed
+
+    Attributes
+    ----------
+    `masker_`: instance of MultiNiftiMasker
+        Masker used to filter and mask data as first step. If an instance of
+        MultiNiftiMasker is given in `mask` parameter,
+        this is a copy of it. Otherwise, a masker is created using the value
+        of `mask` and other NiftiMasker related parameters as initialization.
+
+    `mask_img_`: Niimg-like object
+        See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+        The mask of the data. If no mask was given at masker creation, contains
+        the automatically computed mask.
+
+    `components_`: list of 2D numpy array (n_components x n-voxels)
+        Arrays of masked extracted components, for each subject. They can be unmasked thanks to
+        the `masker_` attribute.
+
+    """
+    def __init__(self, n_components=20, smoothing_fwhm=None, mask=None,
+                 standardize=True, target_affine=None,
+                 target_shape=None, low_pass=None, high_pass=None,
+                 t_r=None, memory=Memory(cachedir=None), memory_level=0,
+                 n_jobs=1, verbose=0,
+                 whiten=False,
+                 random_state=None
+                 ):
+        self.mask = mask
+        self.memory = memory
+        self.memory_level = memory_level
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+
+        self.n_components = n_components
+        self.smoothing_fwhm = smoothing_fwhm
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.standardize = standardize
+        self.random_state = random_state
+
+        self.whiten = whiten
+
+    def fit(self, imgs, y=None, confounds=None):
+        """Compute the mask and the components
+
+        Parameters
+        ----------
+        imgs: list of Niimg-like objects
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            Data on which the PCA must be calculated. If this is a list,
+            the affine is considered the same for all.
+        """
+
+        random_state = check_random_state(self.random_state)
+        # Hack to support single-subject data:
+        if isinstance(imgs, (_basestring, nibabel.Nifti1Image)):
+            imgs = [imgs]
```

Comment:
This is not the most elegant code but if you really need that, you should probably call `check_niimg(imgs, ensure_ndim=4, atleast_4d=True)`.